src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyAloneOrSingleLine2.java

```diff
@@ -0,0 +1,69 @@
+package com.puppycrawl.tools.checkstyle.checks.blocks.rightcurly;
+
+public class InputRightCurlyAloneOrSingleLine2 {
+
+    private int a;
+    private static int b;
+    {  a = 2; }
+    static { b = 3; }
+
+    void method1() {
+        Thread t = new Thread() {@Override public void run() {
+            int a; int b;} //violation
+        };
+    }
+
+    void method2(java.util.HashSet<String> set) {
+        java.util.Map<String, String> map1 = new java.util.LinkedHashMap<String, String>() {{
+            put("Hello", "World");
+            put("first", "second");
+            put("polygene", "lubricants");
+            put("alpha", "betical");}  //violation
+        };
+
+        java.util.Map<String, String> map2 = new java.util.LinkedHashMap<String, String>() {{
+            put("Hello", "World");
+            put("first", "second");
+            put("polygene", "lubricants");
+            put("alpha", "betical");
+        }};; //violation
+    }
+
+    void method3() {
+        method2(new java.util.HashSet<String>() {{
+            add("XZ13s");
+            add("AB21/X");
+            add("YYLEX");
+            add("AR5E");
+        }});  //violation
+    }
+
+    int method4(int a) {
+        if (a>2) a*=10; return ++a; }   //violation
+
+    void method5(int a) {
+        while (a > 5) { a--; }
+
+        if (a > 4) { a++; }
+
+        do {a--;} while (a > 3);  //NO violation
+
+        for (int i = 1; i < 10; i++) { byte b = 10; }
+
+        if (a < 2) { --a; } else if (a > 3) { a++; }
+
+        java.util.List<String> list = new java.util.ArrayList<>();
+        list.stream()
+                //violation
+                .filter(e -> {return !e.isEmpty() && !"null".equals(e);} )
```

Comment:
please make comment as trailing for a line where violation is reported.

Reply:
If violation on lambda, then there won't be one anymore as we just removed this. See 32fed6694b84b5773e74dfd07b2b080a8d498128 .

Reply:
Done. _(Removed the comment.)_